pkg/kubelet/apis/config/types.go

```diff
@@ -656,3 +661,16 @@ type MemorySwapConfiguration struct {
 	// +optional
 	SwapBehavior string
 }
+
+// RateLimitConfiguration is used for setting the rate limits for the kubelet APIs exposed
+// locally on the node as gRPC over UNIX-domain sockets, like the podresources API
+type RateLimitConfiguration struct {
+	// MaxFrequency  defines the maximum allowed frequency of the calls represented
+	// as number of calls per second, cumulating all the calls.
+	// Omitting the value or setting to zero means no limit.
+	MaxFrequency float64
+	// MaxBurst defines the maximum numbers of calls that may happen at once, within
+	// the boundaries set by frequency.
+	// This value is ignored if the frequuency is unlimited.
+	MaxBurst int32
```

Comment:
Looks like MaxBurst should be > 0 if frequency is NOT unlimited, looking at:
https://pkg.go.dev/golang.org/x/time/rate#:~:text=A%20zero%20Burst%20allows%20no%20events%2C%20unless%20limit%20%3D%3D%20Inf.

Reply:
thanks for looking into this, will check and update